CODEOWNERS

```diff
@@ -1,15 +1,17 @@
 # Code owners groups and a brief description of their areas:
-# @cilium-janitors      Catch-all for code not otherwise owned
+# @cilium/janitors      Catch-all for code not otherwise owned
 # @cilium/agent         Cilium Agent
 # @cilium/bpf           BPF Data Path
+# @cilium/build         Building and packaging
 # @cilium/ci            Continuous integration, testing
 # @cilium/cli           Commandline interfaces
-# @cilium/contributing  Developer documentation
+# @cilium/contributing  Developer documentation & tools
 # @cilium/docker        Docker plugin
 # @cilium/docs          Documentation, examples
 # @cilium/endpoint      Endpoint package
 # @cilium/health        Cilium cluster health tool
 # @cilium/hubble        Hubble integration
+# @cilium/ipam          IPAM
```

Comment:
This entry is currently only used for `/pkg/ipam/` and I can't see a breakdown in several IPAM implementations in there. There is `/pkg/azure/ipam/` though that isn't matched by the `CODEOWNERS` currently.
In addition, do we really have different sets of folks maintaining the different IPAM implementations?

Reply:
FWIW the core two contributors who know the most about IPAM today would be @tgraf @ungureanuvladvictor and they've both been involved in EKS and Azure IPAM. I've heard that other cloud IPAM APIs are often similar so I don't think it's unreasonable to have one overall team for IPAM.

Reply:
From the sidelines I'm down to be auto-tagged on all cloud IPAM PRs. Usually the direct routing on cloud providers is the same, just the cloud API interactions is the different thing between them (this is just me saying from experience with Azure/AWS, but I don't expect GCP to be that different).

Up to you folks on what you want to do here but overall I'd like to be tagged on AWS/Azure things if possible given that's my main area of focus inside Cilium.

Reply:
@joestringer @aanm What about a breakdown into:
```
/pkg/ipam/ @cilium/ipam
/pkg/imap/allocator/aws/ @cilium/ipam @cilium/aws
/pkg/imap/allocator/azure/ @cilium/ipam @cilium/azure
/pkg/aws/ @cilium/aws
/pkg/policy/groups/aws/ @cilium/aws
/pkg/azure/ @cilium/azure
```
Downside is it requires three new teams.

Reply:
That looks sane to me. Can you also add `/pkg/ipam/allocator/operator/`? (it's not in master yet)

also, since we don't have `imap` support yet, can you change the `pkg/imap` to `pkg/ipam`?

Reply:
> Can you also add /pkg/ipam/allocator/operator/? (it's not in master yet)

Sure, but what should I add it under?

> also, since we don't have imap support yet, can you change the pkg/imap to pkg/ipam?

What? We don't?! :smile:

Reply:
>Sure, but what should I add it under? @cilium/operator?

yeah I think that's good

Reply:
I updated the branch with that.